### PR TITLE
Error tests.

### PR DESCRIPTION
**Pull request by pq**
@bwilkerson 
@scheglov

### CODE REVIEW

File: test/error_test.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    expect(errorsForFile('void main() => print("Hello, world!");'), isNull);
+  });
+
+  test("an empty Dart file doesn't throw any errors", () {
+    expect(errorsForFile(''), isNull);
+  });
+
+  test("an error on the first line", () {
+    expect(errorsForFile('void foo;\n'),
+        equals("Error in test.dart: Variables cannot have a type of 'void'\n"));
```
Comment (bwilkerson) [inline test/error_test.dart:22]: Do we want these to be
  expect(errorsForFile(...), isNotEmpty);
to protect against changes in error messages? (I'm not sure of the right answer.)

**Comment by bwilkerson** [comment]
LGTM

**Comment by scheglov** [comment]
LGTM

File: test/error_test.dart (hunk 1 above)
Comment (pq) [inline test/error_test.dart:22]: Not sure.  It's a good point.  My leaning is to land these (as a literal port) and then revisit all the tests as we add functionality.  That said, if you do find the right answer in the meantime, I'm all ears! ;)

**Pull request merged**